### PR TITLE
Move custom elements stuff to their own dom::bindings submodule

### LINKED ISSUE #19179: Move custom elements stuff to their own dom::bindings submodule
The functions `html_constructor` etc have nothing to do in `dom::bindings::interface`.

### PR DESCRIPTION
**Pull request by olmanz**
r? jdm 
Moved the following functions from components/script/dom/bindings/interface.rs to a new file components/script/dom/bindings/htmlconstructor.rs:

- html_constructor()
- get_constructor_object_from_local_name()
- pop_current_element_queue()
- push_new_element_queue()

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #19179.

- [X] These changes do not require tests

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19254)

### CODE REVIEW

**Comment by cbrewster** [comment]
```
Checking files for tidiness...

./components/script/dom/bindings/htmlconstructor.rs:1: incorrect license

./components/script/dom/bindings/htmlconstructor.rs:172: trailing whitespace

./components/script/dom/bindings/htmlconstructor.rs:173: Line is longer than 120 characters

./components/script/dom/bindings/htmlconstructor.rs:77: use statement is not in alphabetical order

	expected: html5ever::interface::QualName

	found: html5ever::LocalName

./components/script/dom/bindings/htmlconstructor.rs:78: use statement contains braces for single import

./components/script/dom/bindings/htmlconstructor.rs:80: use statement is not in alphabetical order

	expected: js::jsapi::HandleObject

	found: js::jsapi::{CallArgs, CurrentGlobalOrNull}
```

File: components/script/dom/bindings/htmlconstructor.rs [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+
+            // Step 13
+            DomRoot::downcast(element).ok_or(Error::InvalidState)
+        },
+        // Step 10
+        Some(ConstructionStackEntry::AlreadyConstructedMarker) => Err(Error::InvalidState),
+    }
+}
+                                  
+                                  /// Returns the constructor object for the element associated with the given local name.
```
Comment (jdm) [inline components/script/dom/bindings/htmlconstructor.rs]: nit: remove the extra indentation.

**Review by jdm — CHANGES_REQUESTED** [review]
Once the test-tidy errors are fixed this should be good to merge!

**Comment by olmanz** [comment]
I fixed the tidy-test issues, now it schould be OK.

**Comment by jdm** [comment]
@bors-servo r+
Thanks!

**Comment by olmanz** [comment]
I saw that one of the tests has still failed, but I'm not sure why....
./mach test-tidy has worked, but the test that has failed is ./mach filter-intermittents..... How can this be fixed?

**Comment by KiChjang** [comment]
```
  ▶ Unexpected subtest result in /css/cssom-view/elementsFromPoint-iframes.html:
  └ PASS [expected FAIL] elementsFromPoint on inner documents
```
Is this a legitimate PASS?

**Comment by jdm** [comment]
Can't be; this is just moving code around. I've filed #19273.

**Comment by jdm** [comment]
@bors-servo: retry
* #19273

**Pull request merged**